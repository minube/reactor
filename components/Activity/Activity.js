import React from 'react';
import { ActivityIndicator } from 'react-native';

import { ENV, THEME } from '../../common';

const { IS_SERVER } = ENV;
const { COLOR, UNIT } = THEME;

export default ({ ...inherit }) => (
  <ActivityIndicator
    {...inherit}
    color={COLOR.BORDER}
    style={[
      inherit.style,
      IS_SERVER && { maxHeight: UNIT, maxWidth: UNIT },
    ]}
  />);
