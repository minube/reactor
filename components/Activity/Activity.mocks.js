import { THEME } from '../../common';

const COLOR = THEME.COLOR.ERROR;
const SIZE = 'large';
const STYLE = {
  padding: THEME.SPACE.REGULAR,
  backgroundColor: THEME.COLOR.PRIMARY,
};

export default { COLOR, SIZE, STYLE };
