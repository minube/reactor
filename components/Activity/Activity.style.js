import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const { COLOR, UNIT } = THEME;

const SIZE = UNIT * 0.6;

export default StyleSheet.create({
  container: LAYOUT.STYLE.ROW,

  dot: {
    width: SIZE,
    height: SIZE,
    borderRadius: SIZE / 2,
    backgroundColor: COLOR.PRIMARY,
    marginHorizontal: SIZE / 2,
  },

  large: {
    width: UNIT,
    height: UNIT,
    borderRadius: SIZE,
    marginHorizontal: SIZE / 1.1,
  },
});
