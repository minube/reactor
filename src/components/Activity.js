import React from 'react';
import { ActivityIndicator } from 'react-native';
import { THEME } from '../common';

const { COLOR } = THEME;

export default ({ ...inherit }) => <ActivityIndicator color={COLOR.ACCENT} {...inherit} />;
