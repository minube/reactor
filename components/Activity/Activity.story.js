import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  boolean, color, object, select,
} from '@storybook/addon-knobs/react';

import Activity from './Activity';

const STYLE = {
  padding: 10,
  backgroundColor: 'rgba(0,255,0,0.25)',
};

storiesOf('✅ Activity', module)
  .add('default', () => (
    <Activity />
  ))
  .add('color', () => (
    <Activity color="green" />
  ))
  .add('size', () => (
    <Activity size="large" />
  ))
  .add('style', () => (
    <Activity style={STYLE} />
  ))
  .add('🏀 Playground', () => (
    <Activity
      color={color('color', 'green')}
      size={select('size', ['small', 'large'], 'small')}
      style={object('style', STYLE)}
    />
  ));
