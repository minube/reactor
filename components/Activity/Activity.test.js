import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Activity from './Activity';

describe('<Activity>', () => {
  it('renders', () => {
    const tree = renderer.create(<Activity />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Activity color="#f00" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {size}', () => {
    const tree = renderer.create(<Activity size="large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Activity />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

