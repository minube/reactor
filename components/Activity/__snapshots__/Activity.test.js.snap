// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activity> renders 1`] = `
<ActivityIndicator
  animating={true}
  color="#dbdbdb"
  hidesWhenStopped={true}
  size="small"
  style={
    Array [
      undefined,
      false,
    ]
  }
/>
`;

exports[`<Activity> when {color} 1`] = `
<ActivityIndicator
  animating={true}
  color="#f00"
  hidesWhenStopped={true}
  size="small"
  style={
    Array [
      undefined,
      false,
    ]
  }
/>
`;

exports[`<Activity> when {size} 1`] = `
<ActivityIndicator
  animating={true}
  color="#dbdbdb"
  hidesWhenStopped={true}
  size="large"
  style={
    Array [
      undefined,
      false,
    ]
  }
/>
`;
